server/src/main/java/org/opensearch/cluster/metadata/QueryGroup.java

```diff
@@ -29,30 +29,35 @@
  * Class to define the QueryGroup schema
  * {
  *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga",
- *              "resourceLimits": {
- *                  "jvm": 0.4
+ *              "resource_limits": {
+ *                  "memory": 0.4
  *              },
  *              "resiliency_mode": "enforced",
  *              "name": "analytics",
- *              "updatedAt": 4513232415
+ *              "updated_at": 4513232415
  * }
  */
 @ExperimentalApi
 public class QueryGroup extends AbstractDiffable<QueryGroup> implements ToXContentObject {
 
+    public static final String _ID_STRING = "_id";
+    public static final String NAME_STRING = "name";
+    public static final String RESILIENCY_MODE_STRING = "resiliency_mode";
+    public static final String UPDATED_AT_STRING = "updated_at";
+    public static final String RESOURCE_LIMITS_STRING = "resource_limits";
     private static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
     private final String name;
     private final String _id;
     private final ResiliencyMode resiliencyMode;
     // It is an epoch in millis
     private final long updatedAtInMillis;
-    private final Map<ResourceType, Object> resourceLimits;
+    private final Map<ResourceType, Double> resourceLimits;
 
-    public QueryGroup(String name, ResiliencyMode resiliencyMode, Map<ResourceType, Object> resourceLimits) {
+    public QueryGroup(String name, ResiliencyMode resiliencyMode, Map<ResourceType, Double> resourceLimits) {
```

Comment:
@kaushalmahi12 - I am wondering if this Object -> Double change looks good to you

Reply:
Yeah I talked to him yesterday before making this change

Reply:
Yeah these should be fine. This will us solve the typecasting hassles.